### ISSUE TITLE
New svg layout algorithm broken

### ISSUE THREAD

**Issue by Tarrasch** [spotify/luigi#1301]
The old graph looks like this:

![selection_079](https://cloud.githubusercontent.com/assets/294349/10395674/844577b6-6e9e-11e5-91ab-1a80cd0347ee.png)

and the new one looks like this:

![selection_081](https://cloud.githubusercontent.com/assets/294349/10395679/8cec2d9c-6e9e-11e5-8192-a2af0b5709d4.png)

@daveFNbuck, do you see directly what might have caused this? If not I can try to make it reproducible and  whatnot. I suppose this blocks 2.0 release.

**Comment by daveFNbuck** [comment]
That's a big bug! It looks like every row ended up with 1 node. Not sure why they're all right-oriented. It looks like the graph is wider. Did you make the window wider before generating the second graph? I also see at least one node didn't get laid out, but still somehow has an edge. I don't know why this would happen. Does the bug remain if you revert my complexity fix?

Could you share a pickle file or some code to produce this dependency graph?

**Comment by Tarrasch** [comment]
Hi @daveFNbuck 

Ok, here a starting point:

```python
import luigi

class FinalTask(luigi.Task):
    def requires(self):
        return []

class A(luigi.Task):
    def requires(self):
        yield FinalTask()

class B(luigi.Task):
    def requires(self):
        yield FinalTask()

class Recursive(luigi.Task):
    param = luigi.IntParameter()

    def requires(self):
        if self.param:
            yield Recursive(param=self.param - 1)
        else:
            yield FinalTask()

class Range(luigi.Task):
    def requires(self):
        for i in range(2):
            yield Recursive(param=i)
```

This is how you do it:

    python -m luigi --module hello Range && python -m luigi --module hello A

Then check out the *reverse* dependency graph of `FinalTask`

![selection_089](https://cloud.githubusercontent.com/assets/294349/10399064/ccedee50-6eb0-11e5-8053-3ce5bd19b6aa.png)

Actually you can just start debugging without also launching the `A` task.

![selection_090](https://cloud.githubusercontent.com/assets/294349/10399103/fae4c7c0-6eb0-11e5-86a3-c446ca4d7bbe.png)

The problem here is that they are all on the same line, though they shouldn't be.

**Comment by daveFNbuck** [comment]
They actually should be on the same line. One of the recursive tasks depends on FinalTask, so it's depth 1. The other one depends on the depth 1 task, so it's depth 2. Range depends on both recursive tasks, so it's depth 3. So we have a graph with 4 levels and 1 node on each level.

**Comment by daveFNbuck** [comment]
It turns out there was a bug in the complexity fix for the computeDepth traversal. A more proper fix would compute heights for all nodes on a reversed version of the graph, but this is a bit more annoying to code and requires additional space. I've submitted a PR #1303 to just revert and the graph looks like this

![image](https://cloud.githubusercontent.com/assets/2091885/10401768/f4920eea-6e75-11e5-9b5a-e17ff316e149.png)

**Comment by Tarrasch** [comment]
Fixed by reverting commit in #1303. Thanks Dave!
